src/coreclr/vm/multicorejitimpl.h

```diff
@@ -657,21 +655,32 @@ class MulticoreJitRecorder
 
 public:
 
-    MulticoreJitRecorder(AppDomain * pDomain, ICLRPrivBinder * pBinderContext, bool fAppxMode)
+    MulticoreJitRecorder(AppDomain * pDomain, ICLRPrivBinder * pBinderContext, bool fRecorderActive)
         : m_stats(pDomain->GetMulticoreJitManager().GetStats())
+        , m_ModuleList(nullptr)
+        , m_JitInfoArray(nullptr)
     {
         LIMITED_METHOD_CONTRACT;
 
         m_pDomain           = pDomain;
         m_pBinderContext    = pBinderContext;
+
+        if (fRecorderActive)
+        {
+            m_ModuleList        = new (nothrow) RecorderModuleInfo[MAX_MODULES];
+        }
         m_ModuleCount       = 0;
+
         m_ModuleDepCount    = 0;
 
+        if (fRecorderActive)
+        {
+            m_JitInfoArray      = new (nothrow) RecorderInfo[MAX_METHODS];
```

Comment:
It's possible for these allocations to fail when recording is enabled. It would probably be easiest to refactor these two allocations into a `TryInitialize` function that is called after the constructor, returning false if any allocation failed, and the caller can disable recording in that case.

Reply:
Or could just take a pointer parameter that indicates to the caller whether recording must be disabled

Reply:
Thanks, I've added `CanGatherProfile` function to address this